streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilder.java

```diff
@@ -1008,9 +1020,15 @@ private void buildProcessorNode(final Map<String, ProcessorNode> processorMap,
                         if (internalTopicNames.contains(topic)) {
                             // prefix the internal topic name with the application id
                             final String internalTopic = decorateTopic(topic);
-                            repartitionTopics.put(
+
+                            final RepartitionTopicConfig repartitionTopicConfig = new RepartitionTopicConfig(
                                 internalTopic,
-                                new RepartitionTopicConfig(internalTopic, Collections.emptyMap()));
+                                Collections.emptyMap()
+                            );
+
+                            maybeEnrichRepartitionTopicConfigWithProperties(topic, repartitionTopicConfig);
```

Comment:
Updating `repartitionTopicConfig` in this method may not be the best pattern. Could we pass the number of partitions into the constructor of `RepartitionTopicConfig` instead?